academy/xl-cosmjs/multi-msg.md

```diff
@@ -0,0 +1,508 @@
+        gas: "200000",
+    },
+)
+```
+
+You can confirm this by making the change in your `experiment.ts` from the previous section, and running it again.
+
+<HighlightBox type="tip">
+
+In fact, building a transaction in this way is recommended. `SigningStargateClient` offers you convenience methods such as `sendTokens` for simple use cases only.
+
+</HighlightBox>
+
+## What is this long string?
+
+Note the `typeUrl: "/cosmos.bank.v1beta1.MsgSend"` string. This comes from Protobuf and is a concatenation of:
+
+1. The `package` where `MsgSend` is initially declared:
+
+    ```protobuf [https://github.com/cosmos/cosmos-sdk/blob/3a1027c/proto/cosmos/bank/v1beta1/tx.proto#L2]
+    package cosmos.bank.v1beta1;
+    ```
+
+2. And the name of the message itself, `MsgSend`:
+
+    ```protobuf [https://github.com/cosmos/cosmos-sdk/blob/3a1027c/proto/cosmos/bank/v1beta1/tx.proto#L22]
+    message MsgSend {
+        ...
+    }
+    ```
+
+This is the canonical identifier of the type of the message serialized next to it. Additionally, it is made to be easy for you, the developer, to understand what it represents. The blockchain client knows how to serialize or deserialize it only because this `"/cosmos.bank.v1beta1.MsgSend"` string is passed along. With this `typeUrl`, the blockchain client and CosmJS are able to pick the right deserializer. This object is also named `MsgSend` in `cosmjs-types`.
+
+<HighlightBox type="info">
+
+To learn how to make your own types for your own blockchain project, head to [Create my own CosmJS objects](TODO).
+
+</HighlightBox>
+
+## Multiple token transfer messages
+
+From here, you can add an extra message for a token transfer from Alice to Bob:
+
+```typescript
+const result = await signingClient.signAndBroadcast(
+    alice,
+    [
+        {
+            typeUrl: "/cosmos.bank.v1beta1.MsgSend",
+            value: {
+                fromAddress: alice,
+                toAddress: faucet,
+                amount: [
+                    { denom: "uatom", amount: "100000" },
+                ],
+            },
+          },
+        {
+            typeUrl: "/cosmos.bank.v1beta1.MsgSend",
+            value: {
+                fromAddress: alice,
+                toAddress: bob,
+                amount: [
+                    { denom: "token", amount: "10" },
+                ],
+            },
+          },
+    ],
+    "auto",
+)
+```
+
+## Mixing other message types
+
+The above example shows you two token-transfer messages in a single transaction. You can see this with their `typeUrl: "/cosmos.bank.v1beta1.MsgSend"`.
+
+Neither Cosmos nor CosmJS limit you to messages of the same type. You can decide to have other message types along a token transfer. For instance, in one transaction Alice could:
+
+1. Send tokens to the faucet.
+2. Delegate some of her tokens to Bob the validator.
+
+How would Alice create the second message? Find your message type in `SigningStargateClient`'s registry of types:
```

Comment:
```suggestion
How would Alice create the second message? The `SigningStargateClient` contains a predefined list of `typeUrls` that are supported by default, because they're considered to be the most commonly used messages in the Cosmos SDK. One of them so happens to be `MsgDelegate`, and that's exactly what you'll need. You can click on the source link below to see the rest of the `typeUrls` that come with `SigningStargateClient`:
```